### ISSUE TITLE
Allow outputing completion to stdout

### ISSUE THREAD

**Issue by SuperSandro2000** [ellie/atuin#249]
This is a follow up to #235.

Most other tools output the completion script to stdout all the time or when no directory/file is specified. This allows evaling it in your rc file or in the case of nixpkgs directly install it to the right location without creating a temporary directory first.

Also the command to generate the completion shouldn't rely on `$HOME` being available and create the config file if it does not exist.

**Comment by conradludgate** [comment]
Seems the new v3 version of clap, which we intend to use, has split out the completion generation logic into a separate crate. This does seem to now support writing to stdout instead of a file[0], so we should be able to address this

[0]: https://docs.rs/clap_generate/3.0.0-rc.7/clap_generate/fn.generate.html

**Comment by conradludgate** [comment]
To avoid any unnecessary breaking changes, we can just make the outdir optional. If it's specified we can determine the correct file path automatically and write there, otherwise we can use stdout

**Comment by orhun** [comment]
I suggest we can do something like this after we make `--out-dir` optional:

```sh
# generate completions in the specified directory if --out-dir is specified
$ mkdir completions/
$ atuin gen-completions --shell bash --out-dir completions/

# print to stdout if --out-dir is not specified
$ atuin gen-completions --shell bash

```

**Comment by conradludgate** [comment]
Exactly. I already have a prototype working locally for that. Still depends on clap3 official release but that should be a few weeks away

**Comment by conradludgate** [comment]
Implemented in #289
